packages/test-runner-browserstack/src/browserstackLauncher.ts

```diff
@@ -34,19 +33,35 @@ export class BrowserstackLauncher extends WebdriverLauncher {
       user: capabilities['browserstack.user'] as string,
       key: capabilities['browserstack.key'] as string,
     });
+
+    if (this.capabilities['browserstack.local']) {
+      this.localIp = internalIp.v4.sync() as string;
+      if (!this.localIp) {
+        throw new Error('Can not determine the local IP.');
+      }
+    }
   }
 
   async initialize(config: TestRunnerCoreConfig) {
-    await registerBrowserstackLocal(
-      this,
-      this.capabilities['browserstack.key'] as string,
-      this.localOptions,
-    );
+    if (this.capabilities['browserstack.local']) {
+      await registerBrowserstackLocal(
+        this,
+        this.capabilities['browserstack.key'] as string,
+        this.localOptions,
+      );
+    }
     await super.initialize(config);
   }
 
   startSession(sessionId: string, url: string) {
-    return super.startSession(sessionId, url.replace(/(localhost|127\.0\.0\.1)/, localIp));
+    if (url === 'localhost' || url === '127.0.0.1') {
```

Comment:
Previously this replace these when available, which means `url` likely wasn't exactly `localhost` or `127.0.0.1`, but included them? Be sure that this isn't breaking any functionality on the other side of your change.

Reply:
Good point it is fixed